photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/training/Params.scala

```diff
@@ -25,12 +25,46 @@ import com.linkedin.photon.ml.supervised.TaskType._
 
 /**
+ * Command line arguments for GAME
+ * @param trainDirs Input directories of training data. Multiple input directories are also accepted if they are
+ *                  separated by commas, e.g., inputDir1,inputDir2,inputDir3.
+ * @param trainDateRangeOpt Date range for the training data represented in the form start.date-end.date,
+ *                          e.g. 20150501-20150631. If trainDateRangeOpt is specified, the input directory is expected
+ *                          to be in the daily format structure (e.g., trainDir/daily/2015/05/01), if not, then no
+ *                          assumption is on the structure of the input directory.
+ * @param validateDirsOpt Input directories of validating data. Multiple input directories are also accepted if they
+ *                        are separated by commas, e.g., inputDir1,inputDir2,inputDir3.
+ * @param validateDateRangeOpt Date range for the training data represented in the form start.date-end.date,
+ *                             e.g. 20150501-20150631. If trainDateRangeOpt is specified, the input directory is
+ *                             expected to be in the daily format structure (e.g., validateDir/daily/2015/05/01), if
+ *                             not, then no assumption is made on the structure of the input directory.
+ * @param minPartitionsForValidation Minimum number of partitions for validating data (if provided).
+ * @param featureNameAndTermSetInputPath Input path to the features name-and-term lists.
+ * @param featureShardIdToFeatureSectionKeysMap A map between the feature shard id and it's corresponding feature
+ *                                              section keys in the following format:
+ *                                              shardId1:sectionKey1,sectionKey2|shardId2:sectionKey2,sectionKey3.
+ * @param outputDir Output directory for logs and learned models.
+ * @param numIterations Number of coordinate descent iterations.
+ * @param updatingSequence Updating order of the ordinates (separated by commas) in the coordinate descent algorithm.
+ * @param fixedEffectOptimizationConfigurations Optimization configurations for the fixed effect optimization problem,
+ *                                              multiple configurations are accepted and should be separated by
+ *                                              semi-colon.
+ * @param fixedEffectDataConfigurations Configurations for each fixed effect data set.
+ * @param randomEffectOptimizationConfigurations Optimization configurations for each random effect optimization
+ *                                               problem, multiple parameters are separated by semi-colon.
+ * @param factoredRandomEffectOptimizationConfigurations Optimization configurations for each factored random effect
+ *                                                       optimization problem, multiple parameters are accepted and
+ *                                                       separated by semi-colon.
```

Comment:
Examples might be helpful here. Then again, they're rather large and the integration tests contain examples.